src/json-schema.js

```diff
@@ -5,7 +5,73 @@ function isExtension(value, key) {
   return _.startsWith(key, 'x-');
 }
 
-function convertSubSchema(schema, references) {
+export function parseReference(reference) {
+  const parts = reference.split('/');
+
+  if (parts[0] !== '#') {
+    throw new Error('Schema reference must start with document root (#)');
+  }
+
+  if (parts[1] !== 'definitions' || parts.length !== 3) {
+    throw new Error('Schema reference must be reference to #/definitions');
+  }
+
+  const id = parts[2];
+
+  return id;
+}
+
+function lookupReference(reference, root) {
+  const parts = reference.split('/').reverse();
+
+  if (parts.pop() !== '#') {
+    throw new Error('Schema reference must start with document root (#)');
+  }
+
+  if (parts.pop() !== 'definitions') {
+    throw new Error('Schema reference must be reference to #/definitions');
+  }
```

Comment:
Probably because I am out of context, but I don't understand why we removed the usage of `parseReference` from this function and duplicated the code with minor difference.

Reply:
`parseReference` requires it to be a reference of a definition. Whereas `lookupReference` scans for any arbitary reference inside `definitions`. The latter function is used purely for providing dereferenced values so it isn't strict about the path itself. `lookupReference` is used to copy a reference into the same path so it is very strict about the definition.